src/openrct2-ui/scripting/ScWidget.hpp

```diff
@@ -962,6 +963,15 @@ namespace OpenRCT2::Scripting
                 OpenRCT2::Ui::Windows::SetWidgetMaxLength(w, _widgetIndex, value);
             }
         }
+
+        void focus()
+        {
+            auto w = GetWindow();
+            if (w != nullptr && IsCustomWindow())
+            {
+                w->OnMouseDown(_widgetIndex);
```

Comment:
Have you tried setting focus on a button widget? I think it might actually call a button press event doing this.

Reply:
I did think about that. But buttons are triggered by `OnMouseUp`, and I don't see the reason for simulating a button press as you could always call the callback function. Maybe there are use cases that I'm not aware of.

Reply:
> Have a look at CustomWindow.cpp and check for OnMouseDown. I don't think the proposed solution is the right way.

Sorry, I don't quite understand. Shouldn't this work if I pass a textbox widget to the function?
https://github.com/OpenRCT2/OpenRCT2/blob/c46701983bad7830009562dc73f7bfdbdaa798c6/src/openrct2-ui/scripting/CustomWindow.cpp#L655-L658

Reply:
Your solution causes a mouse down event which it should not do, you have to look a bit deeper to how it decides what has focus, the window code is one of the messier areas so its probably not a straight forward thing.

Reply:
Perhaps `WindowStartTextbox` (`Window.cpp`) is the function you need?